samcli/commands/init/init_templates.py

```diff
@@ -171,3 +184,79 @@ def is_dynamic_schemas_template(self, package_type, app_template, runtime, base_
             if option.get("appTemplate") == app_template:
                 return option.get("isDynamicTemplate", False)
         return False
+
+    def get_preprocessed_manifest(self, filter_value=None):
+        """
+        This method get the manifest cloned from the git repo and preprocessed it.
+        The structure of the manifest is shown below:
+
+        {
+            "dotnetcore2.1": [
+                {
+                    "directory": "dotnetcore2.1/cookiecutter-aws-sam-hello-dotnet",
+                    "displayName": "Hello World Example",
+                    "dependencyManager": "cli-package",
+                    "appTemplate": "hello-world",
+                    "packageType": "Zip",
+                    "useCaseName": "Serverless API"
+                },
+            ]
+        }
+
+        Raises:
+            InvalidInitTemplateError: Exception raise when template is invalid or missing a parameter
+
+        Returns:
+            dict: This is preprocessed manifest with the use_case as key
+        """
+        if not self._git_repo.clone_attempted:
+            self.git_cloner()
+        manifest_path = os.path.join(self._git_repo.local_path, "manifest.json")
```

Comment:
The decision to use `path` was made because this would fail pr test as module `os` has no `Path` member. if you inspect the code base you would notice that `path` was used in place of `Path`

Reply:
`Path` is not part of `os` and is already imported in this file: `from pathlib import Path`